tests/system-containers/main.yml

```diff
@@ -0,0 +1,511 @@
+
+    - name: Install flannel
+      command:
+        atomic install
+        --system
+        --name=flannel
+        {{ flannel_image }}
+
+    - name: Start flannel
+      command: systemctl start flannel
+
+    - name: Verify flannel service is active
+      command: systemctl is-active flannel
+      register: flannel_output
+      failed_when: "'active' not in flannel_output.stdout"
+
+    - include: 'roles/atomic_system_uninstall/tasks/main.yml'
+      vars:
+        name: "{{ etcd_name }}"
+
+    - include: 'roles/atomic_system_uninstall/tasks/main.yml'
+      vars:
+        name: "{{ flannel_name }}"
+
+    # Upstream issue with flannel container installation stopping docker
+    # https://github.com/projectatomic/atomic-system-containers/issues/23
+    - name: Start docker
+      command: systemctl start docker
+      when: "'gscrivano' in etcd_image"
+
+    - include: 'roles/atomic_system_uninstall_verify/tasks/main.yml'
+      vars:
+        image: "{{ etcd_image }}"
+
+    - include: 'roles/atomic_system_uninstall_verify/tasks/main.yml'
+      vars:
+        image: "{{ flannel_image }}"
+
+- name: System Containers - Set RUN_DIRECTORY & STATE_DIRECTORY
+  hosts: all
+  become: yes
+
+  # - Verify setting RUN_DIRECTORY and STATE_DIRECTORY
+
+  tags:
+    - run_state_dir
+
+  vars_files:
+    - vars.yml
+
+  tasks:
+    - name: Set facts
+      set_fact:
+        run_dir: '/root'
+        state_dir: '/root'
+
+    - name: Create container with RUN_DIRECTORY and STATE_DIRECTORY
+      command: >
+        atomic install
+        --system
+        --set=RUN_DIRECTORY={{ run_dir }}
+        --set=STATE_DIRECTORY={{ state_dir }}
+        {{ g_hw_image }}
+
+    - name: Get output of {{ g_hw_name }} info file
+      command:  cat /var/lib/containers/atomic/{{ g_hw_name }}/info
+      register: info_output
+
+    - name: Convert output to JSON
+      set_fact:
+        info_json: "{{ info_output.stdout|from_json }}"
+
+    - name: Verify RUN_DIRECTORY is set
+      fail:
+        msg: "RUN_DIRECTORY is not set to {{ run_dir }}"
+      when: run_dir not in info_json['values']['RUN_DIRECTORY']
+
+    - name: Verify STATE_DIRECTORY is set
+      fail:
+        msg: "STATE_DIRECTORY is not set to {{ state_dir }}"
+      when: state_dir not in info_json['values']['STATE_DIRECTORY']
+
+    - include: 'roles/atomic_system_uninstall/tasks/main.yml'
+      vars:
+        name: "{{ g_hw_name }}"
+
+    - include: 'roles/atomic_system_uninstall_verify/tasks/main.yml'
+      vars:
+        image: "{{ g_hw_image }}"
+
+- name: System Containers - Negative Testing
+  hosts: all
+  become: yes
+
+  #  Negative Testing
+  #    - Verify error message is displayed when uninstalling a system container that does not exist
+  #    - Verify error message is displayed when installing a non system container with the --system option?
+
+  tags:
+    - negative
+
+  vars_files:
+    - vars.yml
+
+  tasks:
```

Comment:
The system containers code will try to make non system containers into system containers by creating the the files necessary to start the services.  It doesn't mean that it will work though.  I tried running atomic install --system with the rhel7 container and it created a rhel7 service.